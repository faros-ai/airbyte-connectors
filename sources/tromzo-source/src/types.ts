export interface TromzoConfig {
  api_key: string;
  organization: string;
  api_timeout?: number;
  api_page_size?: number;
  api_max_retries?: number;
}

export interface Finding {
  // Is this needed?
  id: string;
  repository?: {
    id?: string;
    uid?: string;
  };
  toolName?: string;
  dbCreatedAt?: string;
  dbUpdatedAt?: string;
  scannerCreatedAt?: string;
  scannerUpdatedAt?: string;
  businessRisk?: string;
  dismissReason?: string;
  dismissedAt?: string;
  scannerDismissedAt?: string;
  userDueDate?: string;
  sourceFilename?: string;
  sourcePath?: string;
  vulnerableVersion?: string;
  vulnerability?: {
    severity?: string;
    cve?: string;
    ghsa?: string;
    fixAvailable?: boolean;
  };
  status?: string;
  scannerStatus?: string;
  userUpdatedStatus?: string;
  line?: number;
  jiraUrl?: string;
  lastReviewed?: string;
  scannerConfidence?: string;
  asset?: {
    name?: string;
    id?: string;
    uid?: string;
    type?: string;
    description?: string;
  };
  projects?: any; // Type needs to be specified based on actual data structure
  key?: string;
}
