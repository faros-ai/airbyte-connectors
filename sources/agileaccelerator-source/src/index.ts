import {Command} from 'commander';
import {
  AirbyteLogger,
  AirbyteSourceBase,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import VError from 'verror';

import {
  Agileaccelerator,
  AgileacceleratorConfig,
} from './agileaccelerator/agileaccelerator';
import {Works} from './streams';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteLogger();
  const source = new AgileacceleratorSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

/** AgileAccelerator source implementation. */
export class AgileacceleratorSource extends AirbyteSourceBase<AgileacceleratorConfig> {
  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }
  async checkConnection(
    config: AgileacceleratorConfig
  ): Promise<[boolean, VError]> {
    try {
      const agileaccelerator = await Agileaccelerator.instance(
        config,
        this.logger
      );
      await agileaccelerator.checkConnection(config);
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }
  streams(config: AgileacceleratorConfig): AirbyteStreamBase[] {
    return [new Works(config, this.logger)];
  }
}
