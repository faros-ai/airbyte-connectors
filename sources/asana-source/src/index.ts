import {Command} from 'commander';
import {
  AirbyteSourceBase,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import {AirbyteSourceLogger} from 'faros-airbyte-cdk';
import VError from 'verror';

import {Asana, AsanaConfig} from './asana';
import {
  Projects,
  ProjectTasks,
  Tags,
  Tasks,
  Users,
  Workspaces,
} from './streams';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteSourceLogger();
  const source = new AsanaSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

/** Asana source implementation. */
export class AsanaSource extends AirbyteSourceBase<AsanaConfig> {
  get type(): string {
    return 'asana';
  }

  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }
  async checkConnection(config: AsanaConfig): Promise<[boolean, VError]> {
    try {
      const asana = Asana.instance(config, this.logger);
      await asana.checkConnection();
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }
  streams(config: AsanaConfig): AirbyteStreamBase[] {
    return [
      new Projects(config, this.logger),
      new Tags(config, this.logger),
      new Tasks(config, this.logger),
      new Users(config, this.logger),
      new Workspaces(config, this.logger),
      new ProjectTasks(config, this.logger),
    ];
  }
}
