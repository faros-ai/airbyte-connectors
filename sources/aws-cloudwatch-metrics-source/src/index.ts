import {Command} from 'commander';
import {
  AirbyteSourceBase,
  AirbyteSourceLogger,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import VError from 'verror';

import {CloudWatch, Config} from './cloudwatch';
import {Metrics} from './streams';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteSourceLogger();
  const source = new CloudWatchMetricsSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

/** CloudWatch Metrics source implementation. */
export class CloudWatchMetricsSource extends AirbyteSourceBase<Config> {
  get type(): string {
    return 'cloudwatch-metrics';
  }

  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }
  async checkConnection(config: Config): Promise<[boolean, VError]> {
    try {
      const cloudWatch = CloudWatch.instance(config);
      await cloudWatch.checkConnection();
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }
  streams(config: Config): AirbyteStreamBase[] {
    return [new Metrics(config, this.logger)];
  }
}
