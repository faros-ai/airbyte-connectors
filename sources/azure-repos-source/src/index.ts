import {Command} from 'commander';
import {
  AirbyteLogger,
  AirbyteSourceBase,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import VError from 'verror';

import {AzureRepoConfig, AzureRepos} from './azure-repos';
import {PullRequests, Repositories, Users} from './streams';
import {Commits} from './streams/commits';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteLogger();
  const source = new AzureRepoSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

/** AzureRepo source implementation. */
export class AzureRepoSource extends AirbyteSourceBase<AzureRepoConfig> {
  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }
  async checkConnection(config: AzureRepoConfig): Promise<[boolean, VError]> {
    try {
      const azureRepos = await AzureRepos.make(config, this.logger);
      await azureRepos.checkConnection();
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }
  streams(config: AzureRepoConfig): AirbyteStreamBase[] {
    return [
      new Commits(config, this.logger),
      new PullRequests(config, this.logger),
      new Repositories(config, this.logger),
      new Users(config, this.logger),
    ];
  }
}
