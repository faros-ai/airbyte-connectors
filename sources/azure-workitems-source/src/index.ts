import {Command} from 'commander';
import {
  AirbyteLogger,
  AirbyteSourceBase,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import VError from 'verror';

import {AzureWorkitems, AzureWorkitemsConfig} from './azure-workitems';
import {Boards, Iterations, Users} from './streams';
import {Workitems} from './streams/workitems';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteLogger();
  const source = new AzureWorkitemsSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

/** Example source implementation. */
export class AzureWorkitemsSource extends AirbyteSourceBase<AzureWorkitemsConfig> {
  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }
  async checkConnection(
    config: AzureWorkitemsConfig
  ): Promise<[boolean, VError]> {
    try {
      const azureActiveDirectory = await AzureWorkitems.instance(config);
      await azureActiveDirectory.checkConnection();
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }
  streams(config: AzureWorkitemsConfig): AirbyteStreamBase[] {
    return [
      new Workitems(config, this.logger),
      new Users(config, this.logger),
      new Iterations(config, this.logger),
      new Boards(config, this.logger),
    ];
  }
}
