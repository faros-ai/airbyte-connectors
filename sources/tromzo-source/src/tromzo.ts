import {AxiosInstance} from 'axios';
import {AirbyteLogger} from 'faros-airbyte-cdk';
import {makeAxiosInstanceWithRetry} from 'faros-js-client';
import fs from 'fs';
import path from 'path';
import {Memoize} from 'typescript-memoize';
import VError from 'verror';

import {Finding, TromzoConfig} from './types';

export class Tromzo {
  private static readonly tromzo: Tromzo;
  constructor(
    private readonly api: AxiosInstance,
    private readonly limit: number = 100,
    private readonly logger?: AirbyteLogger
  ) {}

  static async instance(
    config: TromzoConfig,
    logger: AirbyteLogger
  ): Promise<Tromzo> {
    if (Tromzo.tromzo) return Tromzo.tromzo;

    const apiKey = config.api_key?.trim();
    if (!apiKey) {
      throw new VError('Please provide a valid Tromzo API key');
    }
    const organization = config.organization?.trim();
    if (!organization) {
      throw new VError('Please provide a valid Tromzo organization');
    }

    const baseURL = `https://${organization}.tromzo.com/api/`;
    const api = makeAxiosInstanceWithRetry(
      {
        baseURL,
        timeout: config.api_timeout ?? 0, // 0 means no timeout
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${apiKey}`,
        },
        maxContentLength: Infinity, //default is 2000 bytes
      },
      logger?.asPino(),
      config.api_max_retries
    );

    return new Tromzo(api, config.api_page_size, logger);
  }

  async checkConnection(): Promise<void> {
    await this.tools();
  }

  private static readQueryFile(fileName: string): string {
    return fs.readFileSync(
      path.join(__dirname, '..', 'resources', 'queries', fileName),
      'utf8'
    );
  }

  @Memoize()
  async tools(): Promise<string[]> {
    const query = Tromzo.readQueryFile('tool-names.gql');
    try {
      const response = await this.api.post('', {query});
      const tools = response.data?.data?.findings?.toolNames;
      if (!tools?.length) {
        throw new VError('No configured tools found');
      }
      return tools;
    } catch (err) {
      throw new VError(err, 'Failed to fetch tools from Tromzo');
    }
  }

  async *findings(toolName: string, startDate: Date): AsyncGenerator<Finding> {
    const query = Tromzo.readQueryFile('findings.gql');
    let offset = 0;
    let count = 0;
    let totalObjects = 0;

    const q =
      // Tromzo doesn't support db_updated_at for GitHub services, so will fetch all findings
      toolName.toLowerCase().startsWith('github')
        ? `tool_name in ("${toolName}")`
        : `tool_name in ("${toolName}") and db_updated_at >= "${startDate.toISOString()}"`;

    do {
      const variables = {
        offset,
        first: this.limit,
        q,
      };
      let response;

      try {
        response = await this.api.post('', {query, variables});
      } catch (err) {
        throw new VError(err, 'Failed to fetch findings from Tromzo');
      }

      const findings = response.data?.data?.findings;
      const edges = findings?.edges;
      if (!edges?.length) {
        break;
      }

      for (const edge of edges) {
        yield edge.node;
        count++;
      }

      totalObjects = findings?.pageInfo?.totalObjects;
      this.logger?.debug(
        `Fetched ${count} records of ${totalObjects} total findings`
      );

      offset += this.limit;
    } while (offset < totalObjects);
  }
}
