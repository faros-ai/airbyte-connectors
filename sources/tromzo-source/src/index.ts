import {Command} from 'commander';
import {
  AirbyteConfiguredCatalog,
  AirbyteSourceBase,
  AirbyteSourceLogger,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteState,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import VError from 'verror';

import {Findings} from './streams/findings';
import {Tromzo} from './tromzo';
import {TromzoConfig} from './types';
import {calculateDateRange} from 'faros-airbyte-common/common';

export const DEFAULT_CUTOFF_DAYS = 90;

export function mainCommand(): Command {
  const logger = new AirbyteSourceLogger();
  const source = new TromzoSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

export class TromzoSource extends AirbyteSourceBase<TromzoConfig> {
  get type(): string {
    return 'tromzo';
  }

  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }

  async checkConnection(config: TromzoConfig): Promise<[boolean, VError]> {
    try {
      const tromzo = await Tromzo.instance(config, this.logger);
      await tromzo.checkConnection();
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }

  streams(config: TromzoConfig): AirbyteStreamBase[] {
    return [new Findings(config, this.logger)];
  }
  async onBeforeRead(
    config: TromzoConfig,
    catalog: AirbyteConfiguredCatalog,
    state?: AirbyteState
  ): Promise<{
    config: TromzoConfig;
    catalog: AirbyteConfiguredCatalog;
    state?: AirbyteState;
  }> {
    const {startDate, endDate} = calculateDateRange({
      start_date: config.start_date,
      end_date: config.end_date,
      cutoff_days: config.cutoff_days ?? DEFAULT_CUTOFF_DAYS,
      logger: this.logger.info.bind(this.logger),
    });

    return {
      config: {
        ...config,
        startDate,
        endDate,
      } as TromzoConfig,
      catalog,
      state,
    };
  }
}
