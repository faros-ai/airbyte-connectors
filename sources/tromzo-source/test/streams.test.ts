import {
  AirbyteLogLevel,
  AirbyteSourceLogger,
  SyncMode,
} from 'faros-airbyte-cdk';
import {Utils} from 'faros-js-client';
import fs from 'fs-extra';

import * as sut from '../src/index';
import {Tromzo} from '../src/tromzo';

function readTestResourceFile(fileName: string): any {
  return JSON.parse(fs.readFileSync(`test_files/${fileName}`, 'utf8'));
}

function readQueryFile(fileName: string): string {
  return fs.readFileSync(`resources/queries/${fileName}`, 'utf8');
}

describe('streams', () => {
  const logger = new AirbyteSourceLogger(
    // Shush messages in tests, unless in debug
    process.env.LOG_LEVEL === 'debug'
      ? AirbyteLogLevel.DEBUG
      : AirbyteLogLevel.FATAL
  );

  const config = {
    api_key: 'test_api_key',
    organization: 'test',
    tools: ['codeql'],
    startDate: Utils.toDate('2024-01-01'),
  };

  async function testStream(
    tool: string,
    variables: any,
    syncMode: SyncMode = SyncMode.FULL_REFRESH
  ): Promise<void> {
    const postFn = jest.fn().mockResolvedValue({
      data: readTestResourceFile('findings.json'),
    });

    Tromzo.instance = jest.fn().mockImplementation(() => {
      return new Tromzo({post: postFn} as any, undefined, logger);
    });

    const source = new sut.TromzoSource(logger);
    const streams = source.streams(config);
    const stream = streams[0];
    const iter = stream.readRecords(
      syncMode,
      undefined,
      {tool},
      {codeql: {cutoff: new Date('2024-10-13').getTime()}}
    );

    const items = [];
    for await (const item of iter) {
      items.push(item);
    }
    expect(items).toMatchSnapshot();
    expect(postFn).toHaveBeenCalledWith(
      '',
      expect.objectContaining({
        query: expect.stringContaining(readQueryFile('findings.gql')),
        variables,
      })
    );
  }

  test('findings - full refresh', async () => {
    await testStream('codeql', {
      offset: 0,
      first: 100,
      q: `tool_name in ("codeql") and db_updated_at >= "2024-01-01T00:00:00.000Z"`,
    });
  });

  test('findings - incremental', async () => {
    await testStream(
      'codeql',
      {
        offset: 0,
        first: 100,
        q: `tool_name in ("codeql") and db_updated_at >= "2024-10-13T00:00:00.000Z"`,
      },
      SyncMode.INCREMENTAL
    );
  });

  test('findings - github has no timestamp filtering', async () => {
    await testStream('github dependabot', {
      offset: 0,
      first: 100,
      q: `tool_name in ("github dependabot")`,
    });
  });
});
