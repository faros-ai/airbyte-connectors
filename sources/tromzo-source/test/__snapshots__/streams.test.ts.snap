// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`streams findings - full refresh 1`] = `
[
  {
    "asset": {
      "description": "",
      "id": "ast_1",
      "name": "faros-ai/airbyte-connectors",
      "service": "GitHub",
      "type": "Code repository",
    },
    "dbUpdatedAt": "2024-10-17T09:48:43.407288+00:00",
    "dismissReason": "",
    "dismissedAt": "2024-10-17T09:48:43.405061+00:00",
    "dueDate": "2024-10-30T06:55:39+00:00",
    "key": "fin_1",
    "line": 117,
    "projects": "["connectors"]",
    "scannerCreatedAt": "2024-09-30T06:55:39+00:00",
    "scannerDismissedAt": "2024-10-17T09:48:43.405055+00:00",
    "scannerStatus": "closed",
    "sourceFilename": "jira.ts",
    "sourcePath": "https://github.com/faros-ai/airbyte-connectors/blob/main/sources/jira-source/src/jira.ts",
    "status": "closed",
    "toolName": "codeql",
    "vulnerability": {
      "cve": "",
      "fixAvailable": true,
      "ghsa": "",
      "score": 9,
      "severity": "high",
    },
    "vulnerableVersion": "",
  },
  {
    "asset": {
      "description": "",
      "id": "ast_2",
      "name": "faros-ai/faros-community-edition",
      "service": "GitHub",
      "type": "Code repository",
    },
    "dbUpdatedAt": "2024-11-13T00:39:07.533502+00:00",
    "dismissReason": "",
    "dismissedAt": "2024-11-13T00:39:07.533502+00:00",
    "dueDate": "2024-11-14T01:22:04+00:00",
    "key": "fin_2",
    "line": 118,
    "projects": "["community-edition"]",
    "scannerCreatedAt": "2024-11-07T01:22:04+00:00",
    "scannerDismissedAt": "2024-11-13T00:39:07.533502+00:00",
    "scannerStatus": "closed",
    "sourceFilename": "package-lock.json",
    "sourcePath": "https://github.com/faros-ai/faros-community-edition/blob/main/docker-extension/ui/package-lock.json",
    "status": "closed",
    "toolName": "github dependabot",
    "vulnerability": {
      "cve": "CVE-2024-21538",
      "fixAvailable": false,
      "ghsa": "GHSA-3xgq-45jj-v275",
      "score": 9.5,
      "severity": "high",
    },
    "vulnerableVersion": ">= 2.12|^3.0,< 3.0.0",
  },
]
`;

exports[`streams findings - github has no timestamp filtering 1`] = `
[
  {
    "asset": {
      "description": "",
      "id": "ast_1",
      "name": "faros-ai/airbyte-connectors",
      "service": "GitHub",
      "type": "Code repository",
    },
    "dbUpdatedAt": "2024-10-17T09:48:43.407288+00:00",
    "dismissReason": "",
    "dismissedAt": "2024-10-17T09:48:43.405061+00:00",
    "dueDate": "2024-10-30T06:55:39+00:00",
    "key": "fin_1",
    "line": 117,
    "projects": "["connectors"]",
    "scannerCreatedAt": "2024-09-30T06:55:39+00:00",
    "scannerDismissedAt": "2024-10-17T09:48:43.405055+00:00",
    "scannerStatus": "closed",
    "sourceFilename": "jira.ts",
    "sourcePath": "https://github.com/faros-ai/airbyte-connectors/blob/main/sources/jira-source/src/jira.ts",
    "status": "closed",
    "toolName": "codeql",
    "vulnerability": {
      "cve": "",
      "fixAvailable": true,
      "ghsa": "",
      "score": 9,
      "severity": "high",
    },
    "vulnerableVersion": "",
  },
  {
    "asset": {
      "description": "",
      "id": "ast_2",
      "name": "faros-ai/faros-community-edition",
      "service": "GitHub",
      "type": "Code repository",
    },
    "dbUpdatedAt": "2024-11-13T00:39:07.533502+00:00",
    "dismissReason": "",
    "dismissedAt": "2024-11-13T00:39:07.533502+00:00",
    "dueDate": "2024-11-14T01:22:04+00:00",
    "key": "fin_2",
    "line": 118,
    "projects": "["community-edition"]",
    "scannerCreatedAt": "2024-11-07T01:22:04+00:00",
    "scannerDismissedAt": "2024-11-13T00:39:07.533502+00:00",
    "scannerStatus": "closed",
    "sourceFilename": "package-lock.json",
    "sourcePath": "https://github.com/faros-ai/faros-community-edition/blob/main/docker-extension/ui/package-lock.json",
    "status": "closed",
    "toolName": "github dependabot",
    "vulnerability": {
      "cve": "CVE-2024-21538",
      "fixAvailable": false,
      "ghsa": "GHSA-3xgq-45jj-v275",
      "score": 9.5,
      "severity": "high",
    },
    "vulnerableVersion": ">= 2.12|^3.0,< 3.0.0",
  },
]
`;

exports[`streams findings - incremental 1`] = `
[
  {
    "asset": {
      "description": "",
      "id": "ast_1",
      "name": "faros-ai/airbyte-connectors",
      "service": "GitHub",
      "type": "Code repository",
    },
    "dbUpdatedAt": "2024-10-17T09:48:43.407288+00:00",
    "dismissReason": "",
    "dismissedAt": "2024-10-17T09:48:43.405061+00:00",
    "dueDate": "2024-10-30T06:55:39+00:00",
    "key": "fin_1",
    "line": 117,
    "projects": "["connectors"]",
    "scannerCreatedAt": "2024-09-30T06:55:39+00:00",
    "scannerDismissedAt": "2024-10-17T09:48:43.405055+00:00",
    "scannerStatus": "closed",
    "sourceFilename": "jira.ts",
    "sourcePath": "https://github.com/faros-ai/airbyte-connectors/blob/main/sources/jira-source/src/jira.ts",
    "status": "closed",
    "toolName": "codeql",
    "vulnerability": {
      "cve": "",
      "fixAvailable": true,
      "ghsa": "",
      "score": 9,
      "severity": "high",
    },
    "vulnerableVersion": "",
  },
  {
    "asset": {
      "description": "",
      "id": "ast_2",
      "name": "faros-ai/faros-community-edition",
      "service": "GitHub",
      "type": "Code repository",
    },
    "dbUpdatedAt": "2024-11-13T00:39:07.533502+00:00",
    "dismissReason": "",
    "dismissedAt": "2024-11-13T00:39:07.533502+00:00",
    "dueDate": "2024-11-14T01:22:04+00:00",
    "key": "fin_2",
    "line": 118,
    "projects": "["community-edition"]",
    "scannerCreatedAt": "2024-11-07T01:22:04+00:00",
    "scannerDismissedAt": "2024-11-13T00:39:07.533502+00:00",
    "scannerStatus": "closed",
    "sourceFilename": "package-lock.json",
    "sourcePath": "https://github.com/faros-ai/faros-community-edition/blob/main/docker-extension/ui/package-lock.json",
    "status": "closed",
    "toolName": "github dependabot",
    "vulnerability": {
      "cve": "CVE-2024-21538",
      "fixAvailable": false,
      "ghsa": "GHSA-3xgq-45jj-v275",
      "score": 9.5,
      "severity": "high",
    },
    "vulnerableVersion": ">= 2.12|^3.0,< 3.0.0",
  },
]
`;
